### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Amen-Musingarimi**
## What I did:
**Create a class Classroom with the following:**
- [ ] @label instance variable, initialized in the constructor.
- [ ] Setter and getter for @label 

**Create the has-many/belongs-to relationship between Classroom and Student. It has the following should be implementations:**
- [ ] Create the has-many side (a classroom has many students).
- [ ] Create the belongs-to side (a student belongs to a classroom).
- [ ] When adding a student to a classroom it also sets the classroom for the student.
- [ ] When setting the classroom for a student it also adds it to the classrooms' students.

**Create a class Book with the following:**
- [ ] @title and @author instance variables, initialized in the constructor.
- [ ] Setters and getters for instance variables.

**Create a class Rental with the following:**
- [ ] @date instance variable, should be initialized in the constructor.
- [ ] Setter and getter for @date.

**Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental with the following implementations:**
- [ ] Create the has-many side of Book and Rental (a book has many rentals).
- [ ] Create the belongs-to side of Rental and Book (a rental belongs to a book).
- [ ] Create the has-many side of Person and Rental (a person has many rentals).
- [ ] Create the belongs-to side of Rental and Person (a rental belongs to a person).
- [ ] Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
 require_relative 'person'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(age, classroom, name: 'Unknown', parent_permission: true)
     super(age, name: name, parent_permission: parent_permission)
-    @classroom = classroom
+    classroom.add_student(self)
```
Comment (DammyShittu) [inline student.rb]: - [ ] The Student class should have its own distinct method for the other side of the association, just like the Classroom class.

  ```
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
  ```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (DammyShittu) [inline book.rb]: - [ ] The Book class should have a method for the association with the Person class through the rental.

  ```
  def add_rentals(date, person)
    Rental(date, self, person)
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Amen-Musingarimi,

## Highlights
✔️ Association

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] Your Person class file is missing the association with the Book class through Rental. Kindly fix that. I was unable to highlight the Person class since you made no changes to it.

  ```
  def add_rentals(date, book)
    Rental(date, book, self)
  end
  ```

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
Hi @Amen-Musingarimi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Well-implemented code based on the previous reviewer's request 👍 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**